addons/stock_picking_batch/models/stock_picking.py

```diff
@@ -47,6 +81,11 @@ class StockPicking(models.Model):
     def create(self, vals_list):
         pickings = super().create(vals_list)
         for picking, vals in zip(pickings, vals_list):
+            if picking.picking_type_id.auto_batch and not vals.get('immediate_transfer') and not vals.get('batch_id') and picking.move_ids:
```

Comment:
Not sure if it's worthwhile, but you can maybe also add to this if statement: `picking._is_auto_batchable()` and make the function also check if a single picking is batchable (e.g. if it has too many lines and can never be batched already) so we can avoid looking for/trying to create a batch for a picking can never be auto-batched.

Reply:
Might save some time for those pickings indeed, and I don't think `_is_auto_batchable()` cost much in comparison of the latter searches.
Done.